python/paddle/fluid/dygraph/dygraph_to_static/utils.py

```diff
@@ -407,9 +408,24 @@ def recover_globals_attribute(src_obj, dst_obj):
             dst_globals[k] = v
 
 
+def func_to_source_code(function, dedent=True):
+    """
+    Transforms function into raw string of source code.
+    """
+    if not (inspect.isfunction(function) or inspect.ismethod(function)):
+        raise TypeError(
+            "The type of 'function' should be a function or method, but received {}.".
+            format(type(function).__name__))
+    source_code = inspect.getsource(function)
+    if dedent:
+        source_code = textwrap.dedent(source_code)
+
+    return source_code
+
+
```

Comment:
function `func_to_source_code` only called in `convert_to_static_with_cache`, how about moving it to file `program_translator.py`?

Reply:
`func_to_source_code` is a common function so I put it into `utils.py`. If put it into `program_translator.py` and used in the future, it may lead to cycle import problem.